packages/tasit-sdk/src/TasitSdk.test.js

```diff
@@ -145,7 +170,81 @@ describe("Decentraland", () => {
       );
       await executeOrder.waitForNonceToUpdate();
 
-      await confirmBalances(estate, [buyer.address, seller.address], [1, 0]);
+      await confirmBalances(
+        estate,
+        [buyer.address, seller.address],
+        [1, estateIds.length - 1]
+      );
+    });
+
+    describe("Decentraland tasit app test cases", () => {
+      beforeEach(
+        "create sell orders and remove wallets from contracts",
+        async () => {
+          marketplace.setWallet(seller);
+          for (let assetId of estateIds) {
+            const priceInWei = ONE.toString();
+            const expireAt = Date.now() + duration.years(1);
+            const createOrder = marketplace.createOrder(
+              estate.getAddress(),
+              assetId,
+              priceInWei,
+              expireAt,
+              gasParams
+            );
+            await createOrder.waitForNonceToUpdate();
+          }
+
+          mana.removeWallet();
+          land.removeWallet();
+          estate.removeWallet();
+          marketplace.removeWallet();
+        }
+      );
+
+      it("should list marketplace estates sell orders (without wallet)", async () => {
+        const estates = [];
+        for (let id of estateIds) {
+          estates.push({
+            id,
+            name: await estate.getMetadata(id),
+          });
+        }
+
+        expect(estates[0].name).to.equal("cool estate 0x1");
+      });
+
+      it("should get an estate info (without wallet)", async () => {
+        const estateId = 5;
+        const name = await estate.getMetadata(estateId);
+        expect(name).to.equal("cool estate 0x5");
+      });
+
+      it("should buy an estate", async () => {
+        await manaFaucetTo(ephemeral, TEN);
+        await etherFaucetTo(ephemeral, ONE);
+
+        mana.setWallet(ephemeral);
+        const marketplaceApproval = mana.approve(
```

Comment:
One cool integration test could be a user approving funds for their in-app ephemeral account from their "main mobile wallet address", and then approving the approved funds for the marketplace to control from the ephemeral account

Reply:
That would belong in a separate test, though

Reply:
So, we'll have `ephemeral` and `mainMobileWallet` (with ethers and mana) and `mainMobileWallet`  will approve `ephemeral` to use their funds? Is it right?
`ephemeral` still need some ethers to pay for gas, right?

Reply:
Yeah, exactly! But that can wait for `0.2.0` if you like.

And yeah, until we add some fancy features around paying for gas in tokens by using a metatransaction, the ephemeral account will need some ETH

Reply:
I actually don't know off the top of my head if an approval of an approval works, but I assume it does

Reply:
https://github.com/tasitlabs/TasitSDK/issues/179

Reply:
Great, thanks for creating that issue. I'll resolve it here.